langchain-core/src/runnables/base.ts

```diff
@@ -1912,12 +1934,21 @@ export class RunnableSequence<
             ),
           })
         );
-        nextStepInput = await raceWithSignal(promise, options?.signal);
+        nextStepInput = await raceWithSignal(promise, config);
+        if (!isAsyncIterable(nextStepInput) && checkAbortSignal(config)) {
```

Comment:
Are there other cases that might indicate an active async task? this should cover async iterables, async generators, and `ReadableStream`.